src/Compilers/Server/VBCSCompiler/ServerDispatcher.cs

```diff
@@ -287,35 +285,37 @@ private void HandleCompletedConnections()
                 _connectionList.RemoveAt(i);
                 processedCount++;
 
-                var connectionData = current.Result;
-                ChangeKeepAlive(connectionData.KeepAlive);
-
-                switch (connectionData.CompletionReason)
+                var completionData = current.Result;
+                switch (completionData.Reason)
                 {
-                    case CompletionReason.CompilationCompleted:
-                    case CompletionReason.CompilationNotStarted:
```

Comment:
Previously `CompilationNotStarted` wasn't treated as a server shutdown event. Upon reflection during this change that is simply a bug. It is no different than any other client disconnection scenario. Once I treated that as a shutdown event having all of the other states seemed a bit odd. Decided to collapse into "error" and "not error" here.